Update Scenario0_Tests.xaml.cs

RuilinZhuIntel opened the pull request:
1. Added new test case MagInterference;
2. Added new test case StaticAccuracy;
3. Added new sensor Relative Orientation ;
4. Modified the CreateTestButton function: can set Button Font size in its argument, to handle the case that the test case name very long.
5. Added new function CalculateMagInterference.
6. Added some useful function such as innerProduct and norm, Angle4 and so on.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/busiotools?pullRequest=29) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/busiotools?pullRequest=29) before we can accept your contribution.<br/><br/>:x: RuilinZhuIntel [sign now](https://cla.opensource.microsoft.com/Microsoft/busiotools?pullRequest=29)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/busiotools?pullRequest=29) it.</sub>

jianliu617 commented:
@jialzhu Could you please review this new pull request and merge if it's acceptable? Thank you.

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -329,8 +337,14 @@ private void DisplayPrecondition()
+                    break;
+                case "MagInterference":
+                    instruction.Text = "Please moving approximately 1G magnet pass stationary device in about 0.25m / s\n" +
+                    "you have 30 seconds to do the test.";
```
Suggest to change the instruction to: 
"Please move a magnet (~1G) pass the stationary device at a speed of approximately 0.25 m/s ."
"You have 30 seconds to complete the test."

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
+        }
+        private double norm(double[] quaternion)
+        {
+            double SumOfSquares = 0;
```
rename to sumOfSquares

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
-
             return eulerAngle;
         }
+        private double innerProduct(double[] quaternionA, double[] quaternionB)
```
rename to InnerProduct

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
+            inner += quaternionA[3] * quaternionB[3];
+            return inner;
+        }
+        private double norm(double[] quaternion)
```
rename to Norm

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
+            if (Math.Abs(result) >= 30 && Math.Abs(result) <= 60) result = Math.Abs(45 - result);
+            return result;
+        }
+        private async void staticAccuracyHandler()
```
rename to StaticAccuracyHandler

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
+                for (int count = 10; count >= 0; count--)
+                {
+                    if (i == 0 || i == 5 || i == 10 || i == 14) instruction.Text = "You have" + " " + count + " " + 
+                            "seconds to Rotate the device on a level surface with its y axis pointing to magnetic north...";
```
"seconds to rotate the device on a level surface with its y axis pointing to magnetic north..."

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
+                    if (i == 0 || i == 5 || i == 10 || i == 14) instruction.Text = "You have" + " " + count + " " + 
+                            "seconds to Rotate the device on a level surface with its y axis pointing to magnetic north...";
+                    else if (i < 5) instruction.Text = "You have" + " " + count + " " + 
+                            "seconds to Rotate 90 degrees counterclockwise around the z axis and stop to a static position...";
```
"seconds to rotate the device 90 degrees counterclockwise around the z axis and stop at a static position..."

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
+                    else if (i < 5) instruction.Text = "You have" + " " + count + " " + 
+                            "seconds to Rotate 90 degrees counterclockwise around the z axis and stop to a static position...";
+                    else if (i < 10) instruction.Text = "You have" + " " + count + " " + 
+                            "seconds to Rotate 90 degrees counterclockwise around the x axis and stop to a static position...";
```
"seconds to rotate the device 90 degrees counterclockwise around the x axis and stop at a static position..."

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
+                    else if (i < 10) instruction.Text = "You have" + " " + count + " " + 
+                            "seconds to Rotate 90 degrees counterclockwise around the x axis and stop to a static position...";
+                    else instruction.Text = "You have" + " " + count + " " + 
+                            "seconds to Rotate 90 degrees counterclockwise around the y axis and stop to a static position...";
```
"seconds to rotate the device 90 degrees counterclockwise around the y axis and stop at a static position..."

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -1435,20 +1558,96 @@ private double[] QuaternionToEulerAngle(SensorQuaternion quaternion)
+            }
+            countdown = new Countdown(testLength[testType], testType);
+            startTime = DateTime.Now;
+            instruction.Text = testType + " Is ending...";
```
"is ending..."

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -30,17 +30,20 @@ public enum Directions { left, right, up, down, nothing }
         private List<int> indices;
         private string testType;
         private List<double[]> dataList;
+        private List<double> OriAngles;
```
rename to oriAngles

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -329,8 +337,14 @@ private void DisplayPrecondition()
                                        "Keep it in stationary state.";
                     break;
+                case "StaticAccuracy":
+                    instruction.Text = "Put device on a level surface with (Y axis) pointing to the magnet north.";
```
"Put device on a level surface with its Y axis pointing to the magnet north."

jialzhu commented on sensors/Tools/SensorExplorer/code/SensorExplorer/Scenario0_Tests.xaml.cs:
```diff
@@ -341,21 +355,18 @@ private void AddPivotItem(int sensorType, int index, int totalIndex)
             Button packetLossTestButton = CreateTestButton("Packet Loss Test");
-       
+            Button staticAccuracyButton = CreateTestButton("Static Accuracy Test");
+            Button MagInterferenceButton = CreateTestButton("MagInterference Test", 28);
```
rename to "magInterferenceButton"

RuilinZhuIntel commented:
Hello jialzhu, thanks for the comments, I've renamed all the variables and function names you mentioned and changed some of the instructions based on your advice in the latest commit. Please have a review. Thanks!

jialzhu commented:
Hi Ruilin,
The changes look good. I just merged it into the master branch.
Thanks,
Lisa
